### PR TITLE
Test rspec and capybara

### PR DESCRIPTION
**Pull request by huaphuoc**
Rspec post model with factory girl.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+
+  it "has a valid factory" do
+  	#person = FactoryGirl.build(:person) it is also valid
```
Comment (tran-hai-mulodo) [inline spec/models/post_spec.rb]: @huaphuoc 
Please remove unuse code for this line:
`#person = FactoryGirl.build(:person) it is also valid`
Thanks.

File: Gemfile [hunk 2]
```diff
@@ -39,12 +39,15 @@ gem 'devise'
 gem 'will_paginate'
 gem "paperclip"
 gem 'mini_magick'
+gem 'faker'
```
Comment (tran-hai-mulodo) [inline Gemfile]: @huaphuoc 
We don't need to use 'faker' gem because 'ffaker' is a rewrite of 'faker'.

File: spec/factories/post.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require 'ffaker'
+require 'faker'
```
Comment (tran-hai-mulodo) [inline spec/factories/post.rb]: @huaphuoc
We don't need to require 'faker' library.

File: spec/models/post_spec.rb (hunk 1 above)
Comment (huaphuoc) [reply spec/models/post_spec.rb]: I removed it, thanks.

File: Gemfile (hunk 2 above)
Comment (huaphuoc) [reply Gemfile]: I removed it, thanks.

File: spec/factories/post.rb (hunk 3 above)
Comment (huaphuoc) [reply spec/factories/post.rb]: I removed it, thanks.

File: spec/factories/post.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+FactoryGirl.define do
+  factory :post do
+    title { FFaker::Book.title }
+    content { FFaker::Book.description }
+    quantity { Faker::Number.between(1, 100)}
```
Comment (tran-hai-mulodo) [inline spec/factories/post.rb]: @huaphuoc 
We can use FFaker::Random.rand(1..100) instead of Faker::Number.between(1, 100).

Comment (huaphuoc) [reply spec/factories/post.rb]: Thanks, I fixed it.

File: spec/models/post_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+  it "is invalid without a quantity" do
+    post = FactoryGirl.build(:post, quantity: nil)
+    expect(post).not_to be_valid
+  end
+
+  it "is invalid if Title is too short (minimum is 5 characters)" do
+    post = FactoryGirl.build(:post, title: "Post title 1", content: "Post content 1", quantity: 1)
+    expect(post.title.length).not_to be < 5
+  end
```
Comment (tran-hai-mulodo) [inline spec/models/post_spec.rb]: @huaphuoc 
In the context of block 'it':
` "is invalid if Title is too short (minimum is 5 characters)" do`
We expect the post is invalid if Title is too short. So, we'll prepare data corresponding to the context above:
```
post = FactoryGirl.build(:post, title: "tit", content: "Post content 1", quantity: 1)
expect(post).not_to be_valid
```
Thanks.

File: spec/models/post_spec.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+  it "is invalid if Content is too short (minimum is 10 characters)" do
+    post = FactoryGirl.build(:post, title: "Post title 1", content: "Post content 1", quantity: 2)
+    expect(post.content.length).not_to be < 10
+  end
```
Comment (tran-hai-mulodo) [inline spec/models/post_spec.rb]: @huaphuoc 
In the context of block 'it':
` "is invalid if Content is too short (minimum is 10 characters)" do`
We expect the post is invalid if Content is too short. So, we'll prepare data corresponding to the context above:
```
post = FactoryGirl.build(:post, title: "Post title 1", content: "Post cont", quantity: 2)
expect(post).not_to be_valid
```
Thanks.

File: spec/models/post_spec.rb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+  it "is invalid if Title has already been taken" do
+    post = FactoryGirl.build(:post, title: "Post title 2", content: "Post content 1", quantity: 3)
+    expect(post.title).not_to be("Post title 1")
+  end
```
Comment (tran-hai-mulodo) [inline spec/models/post_spec.rb]: @huaphuoc 
In the context of block 'it':
` "is invalid if Title has already been taken" do`
We expect the post is invalid if Title has already been taken. So, we'll prepare data corresponding to the context above:
```
exist_post = FactoryGirl.create(:post, title: "Post title 1", content: "Post content 1", quantity: 3)
new_post = FactoryGirl.build(:post, title: "Post title 1", content: "Post content 2", quantity: 3)
expect(new_post).not_to be_valid
```
Thanks.

File: spec/models/post_spec.rb [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+  it "is invalid if Quantity is not integer" do
+    post = FactoryGirl.build(:post, title: "Post title 1", content: "Post content 1", quantity: 4)
+    expect(post.quantity).to be_an(Integer)
+  end
```
Comment (tran-hai-mulodo) [inline spec/models/post_spec.rb]: @huaphuoc 
In the context of block 'it':
` "is invalid if Quantity is not integer" do`
We expect the post is invalid if Quantity is not integer. So, we'll prepare data corresponding to the context above:
```
post = FactoryGirl.build(:post, title: "Post title 1", content: "Post content 1", quantity: 'abc')
expect(post).not_to be_valid
```
Thanks.

File: spec/models/post_spec.rb (hunk 5 above)
Comment (huaphuoc) [reply spec/models/post_spec.rb]: Ok thanks.

File: spec/models/post_spec.rb (hunk 6 above)
Comment (huaphuoc) [reply spec/models/post_spec.rb]: Ok thanks.

File: spec/models/post_spec.rb (hunk 7 above)
Comment (huaphuoc) [reply spec/models/post_spec.rb]: Ok thanks.

File: spec/models/post_spec.rb (hunk 8 above)
Comment (huaphuoc) [reply spec/models/post_spec.rb]: Ok thanks.

**Comment by huaphuoc** [comment]
I fixed post model spec.

**Comment by tran-hai-mulodo** [comment]
@huaphuoc 
It's OK. I merged. Thanks.

**Pull request merged**